File: vlc-android/src/org/videolan/vlc/gui/CompatErrorActivity.java
package org.videolan.vlc.gui;

import android.app.Activity;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import org.videolan.libvlc.util.VLCUtil;
import org.videolan.vlc.R;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class CompatErrorActivity extends Activity {
    public final static String TAG = "VLC/CompatErrorActivity";

    /**
     * Simple friendly activity to tell the user something's wrong.
     *
     * Intent parameters (all optional):
     * runtimeError (bool) - Set to true if you want to show a runtime error
     *                       (defaults to a compatibility error)
     * message (string) - the more detailed problem
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.not_compatible);

        String errorMsg = VLCUtil.getErrorMsg();
        if(getIntent().getBooleanExtra("runtimeError", false))
            if(getIntent().getStringExtra("message") != null) {
                errorMsg = getIntent().getStringExtra("message");
                TextView tvo = (TextView)findViewById(R.id.message);
                tvo.setText(R.string.error_problem);
            }

        TextView tv = (TextView)findViewById(R.id.errormsg);
        tv.setText(getResources().getString(R.string.error_message_is) + "\n" + errorMsg);
    }
}

File: vlc-android/src/org/videolan/vlc/gui/CompatErrorActivity.kt
package org.videolan.vlc.gui

import android.app.Activity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import org.videolan.libvlc.util.VLCUtil
import org.videolan.vlc.R

class CompatErrorActivity : Activity() {

    /**
     * Simple friendly activity to tell the user something's wrong.
     *
     * Intent parameters (all optional):
     * runtimeError (bool) - Set to true if you want to show a runtime error
     * (defaults to a compatibility error)
     * message (string) - the more detailed problem
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.not_compatible)

        var errorMsg = VLCUtil.getErrorMsg()
        if (intent.getBooleanExtra("runtimeError", false))
            if (intent.getStringExtra("message") != null) {
                errorMsg = intent.getStringExtra("message")
                val tvo = findViewById<View>(R.id.message) as TextView
                tvo.setText(R.string.error_problem)
            }

        val tv = findViewById<View>(R.id.errormsg) as TextView
        tv.text = resources.getString(R.string.error_message_is) + "\n" + errorMsg
    }

    companion object {
        const val TAG = "VLC/CompatErrorActivity"
    }
}
